### PR TITLE
Updates to better pin cryptography builds for jammy to not need 1.76 rustc

### PR DESCRIPTION
**Pull request by addyess**
### Overview

Pinning `cryptography` versions for building on jammy

### Details

Builds of this charm were failing on jammy containers due to the version of rustc being 1.75, and the `cryptography` library needing a 1.76 rustc compiler to install into the charm. 

Additional changes:
* Update the CI to function with newer additions
* pin the charmcraft channel to 2.x/stable
* test against a more modern microk8s

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -67,17 +65,10 @@ commands =
...
-    # renovate: datasource=pypi
-    pytest-operator==0.27.0
-    # renovate: datasource=pypi
-    tenacity==8.2.2
-    # renovate: datasource=pypi
-    requests==2.28.2
+    pytest-operator
+    lightkube
+    lightkube-models
+    cryptography
```
Comment (mateoflorido) [inline tox.ini:71]: Should we pin `cryptography` here as well?

Comment (addyess) [reply tox.ini:71]: in tox for integration tests -- we get a binary!!  no one has to compile something nutty.  It just works

Comment (mateoflorido) [reply tox.ini:71]: OH, that's great!

**Review by mateoflorido — APPROVED** [review]
LGTM!

**Pull request merged**